Makefile

```diff
@@ -1,48 +1,72 @@
-AWS_REGION ?= us-west-2
-
-# Defaults to Amazon Linux 2 LTS AMI
-# * use the us-west-2 minimal hvm image
-# https://aws.amazon.com/amazon-linux-2/release-notes/
-SOURCE_AMI_ID ?= $(shell aws \
-	--region $(AWS_REGION) \
-	ec2 describe-images \
-	--output text \
-	--filters \
-		Name=owner-id,Values=137112412989 \
-		Name=virtualization-type,Values=hvm \
-		Name=root-device-type,Values=ebs \
-		Name=name,Values=amzn2-ami-minimal-hvm-* \
-		Name=architecture,Values=x86_64 \
-		Name=state,Values=available \
-	--query 'max_by(Images[], &CreationDate).ImageId')
-
-.PHONY: all validate ami 1.12 1.11 1.10
-
-all: 1.12
+DOCKER_VERSION ?= 18.06
+KUBERNETES_BUILD_DATE ?= 2019-03-27
+CNI_VERSION ?= v0.6.0
+CNI_PLUGIN_VERSION ?= v0.7.5
+ARCH ?= x86_64
+BINARY_BUCKET_NAME ?= amazon-eks
+SOURCE_AMI_OWNERS ?= 137112412989
 
+ifeq ($(ARCH), arm64)
+INSTANCE_TYPE ?= a1.large
+else
+INSTANCE_TYPE ?= m4.large
+endif
+
+DATE ?= $(shell date +%Y-%m-%d)
+
+AWS_DEFAULT_REGION ?= us-west-2
+
+T_RED := \e[0;31m
+T_GREEN := \e[0;32m
+T_YELLOW := \e[0;33m
+T_RESET := \e[0m
+
+.PHONY: all
+all: 1.10 1.11 1.12
+
+.PHONY: validate
 validate:
-	packer validate eks-worker-al2.json
+	packer validate \
+		-var instance_type=$(INSTANCE_TYPE) \
+		eks-worker-al2.json
 
-1.10: validate
+.PHONY: k8s
+k8s: validate
+	@echo "$(T_GREEN)Building AMI for version $(T_YELLOW)$(VERSION)$(T_GREEN) on $(T_YELLOW)$(ARCH)$(T_RESET)"
+	$(eval SOURCE_AMI_ID := $(shell aws ec2 describe-images \
```

Comment:
@mcrute I don't follow why this is necessary.  Shouldn't packer's `source_ami_filter` be able to choose based on the architecture passed in [here](https://github.com/awslabs/amazon-eks-ami/blob/master/eks-worker-al2.json#L34)?

Doesn't doing this force the `Makefile` to always force using the `source_ami` instead?  https://github.com/awslabs/amazon-eks-ami/blob/master/eks-worker-al2.json#L30

Could you provide some additional insight here?